src/SFA.DAS.EmployerCommitmentsV2.Web/Services/EmployerAccountsService.cs

```diff
@@ -0,0 +1,48 @@
+﻿using SFA.DAS.EAS.Account.Api.Client;
+using SFA.DAS.EAS.Account.Api.Types;
+using SFA.DAS.EmployerCommitmentsV2.Web.Models.Cohort;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.EmployerCommitmentsV2.Web.Services
+{
+    public class EmployerAccountsService : IEmployerAccountsService
+    {
+        private readonly IAccountApiClient _accountsApiClient;
+
+        public EmployerAccountsService(IAccountApiClient accountApiClient)
+        {
+            _accountsApiClient = accountApiClient;
+        }        
+
+        public async Task<List<LegalEntity>> GetLegalEntitiesForAccount(string accountId)
+        {
+            var listOfEntities = await _accountsApiClient.GetLegalEntitiesConnectedToAccount(accountId);
+
+            var list = new List<LegalEntity>();
+            if (listOfEntities.Count == 0) return list;
+
+            list.AddRange(listOfEntities.Select(async entity => await _accountsApiClient.GetLegalEntity(accountId, Convert.ToInt64(entity.Id))).Select(legalEntityViewModel => new LegalEntity
```

Comment:
could make this parallel to improve the performance although would require some though as some accounts have >900 legal entities. Might be worth raising as a tech debt ticket.

Reply:
tech debt raised : https://skillsfundingagency.atlassian.net/browse/CON-3606